Inject gtag script for user data.

abdelmalekkkkk opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #11015

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

techanvil reviewed (changes requested):
Hey @abdelmalekkkkk, this is looking good. I've left a few minor comments, please take a look.

Additionally, although the QAB is OK, a more thorough way to test this would be to use [Tag Assistant](https://tagassistant.google.com/) to connect to the test site and verify that `user_data`  was pushed to the data layer as expected:

<img width="1032" height="758" alt="image" src="https://github.com/user-attachments/assets/4bc431bc-2583-40b8-89be-80f7c7641273" />

abdelmalekkkkk commented:
Thanks for the review @techanvil. I addressed your comments and added the cleanup to `tear_down`. I also went ahead and removed the redundant cleanup from other tests since it was a quick search and replace. Please take another look and let me know if all is good.

abdelmalekkkkk commented:
@techanvil I also updated the QAB as you suggested.

techanvil commented on includes/Modules/Ads/Enhanced_Conversions.php:
```diff
@@ -66,8 +71,18 @@ protected function get_user_data() {
 	 * Conditionally enqueues the necessary script for Enhanced Conversions.
 	 *
 	 * @since 1.159.0
+	 * @since n.e.x.t Add the hashed user data to gtag if it exists
```
Please end this comment with a period.

```suggestion
	 * @since n.e.x.t Add the hashed user data to the GTag if it exists.
```

This should be enforced by a PHPCS linting rule, to align with our JS linting.

I've made a note to create an issue for this. Alternatively, feel free to raise the issue yourself - please let me know if you do so I can remove it from my to-do list.

abdelmalekkkkk replied:
I created an issue for that #11293

techanvil commented on tests/phpunit/includes/TestCase.php:
```diff
@@ -80,6 +80,11 @@ public function tear_down() {
 		// Clear screen related globals.
 		unset( $GLOBALS['current_screen'], $GLOBALS['taxnow'], $GLOBALS['typenow'] );
+
+		// Clean up scripts hooks to avoid interefence between tests
```
As above, plus there's a typo:

```suggestion
		// Clean up scripts hooks to avoid interference between tests.
```

techanvil reviewed (changes requested):
Thanks @abdelmalekkkkk!

This is nearly there. There are a couple of last things to address, please see my comments.

Additionally - the E2E tests have all failed in CI, due to what looks like a random network error. There were also a couple of failed PHPUnit test runs, which I have successfully re-run (I noticed you did try to re-run them yourselves a couple of times, there must have been something up with the test runners/related infrastructure at the time).

In general, please ensure that all CI workflows have run successfully before moving an issue to CR. This speeds up the CR process. If you run into problems with the CI tasks, either wait a while and try again, flag it on Slack, raise an issue, or, if it's non-critical, simply mention it on the PR when moving it to CR.

<img width="1377" height="1121" alt="image" src="https://github.com/user-attachments/assets/585154a1-715d-46b3-bbf3-1636b3e5446a" />
